base/system/smss/sminit.c

```diff
@@ -1819,9 +1819,8 @@ SmpCreateDynamicEnvironmentVariables(VOID)
     if (NT_SUCCESS(Status))
     {
         /* To combine it into a single string */
-        swprintf((PWCHAR)PartialInfo->Data + wcslen((PWCHAR)PartialInfo->Data),
-                 L", %S",
-                 PartialInfo2->Data);
+        wcsncat((PWCHAR)PartialInfo->Data, L", ",2 * sizeof(WCHAR));
+        wcsncat((PWCHAR)PartialInfo->Data, (PWCHAR)PartialInfo2->Data, ResultLength);
```

Comment:
You should not use `ResultLength` , but instead `PartialInfo2->DataLength` (which is in bytes, so if the corresponding data is a UNICODE string, you would need to use `PartialInfo2->DataLength / sizeof(WCHAR)` for the length in characters).

Reply:
Done